### PR TITLE
Periodically update advisories

### LINKED ISSUE #15: Ensure dependency vulnerability database is updated on a regular cadence
Right now the recipe goes off a [static snapshot](https://github.com/openrewrite/rewrite-java-dependencies/blob/ead72908e86629d55e6dbbfbee80bf52cf410754/src/main/resources/advisories.csv) of the github security advisory database. We need an automatic process, or a regularly repeated manual process, which keeps this database up to date

### PR DESCRIPTION
**Pull request by timtebeek**
## What's changed?
Added a GitHub Action to periodically update the GitHub advisories CSV.
Rework ParseAdvisories to take in two arguments; add test.
Add minimal script called from GitHub Actions to update the advisories CSV.

## What's your motivation?
Fixes #15

## Have you considered any alternatives or workarounds?
Had initially thought to keep this in a separate repository, but after discussion decided against that for now for simplicity.

### CODE REVIEW

File: .github/workflows/advisories.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+name: Update the GitHub advisories CSV
+
+on:
+  pull_request:
+    branches:
+      - main
```
Comment (timtebeek) [inline .github/workflows/advisories.yml:7]:
```suggestion
```

**Comment by timtebeek** [comment]
Works as per: https://github.com/openrewrite/rewrite-java-dependencies/pull/20

File: src/main/java/org/openrewrite/java/dependencies/github/ParseAdvisories.java [hunk 2]
```diff
@@ -40,50 +39,93 @@
...
+    }
+
+    static void parseAdvisories(File advisoriesRepoInput, File advisoriesCsvOutput) throws IOException {
+        try (FileOutputStream fos = new FileOutputStream(advisoriesCsvOutput)) {
+            Files.walkFileTree(advisoriesRepoInput.toPath(), emptySet(), 16, new PathSimpleFileVisitor(fos));
+        }
+    }
+}
+
+class PathSimpleFileVisitor extends SimpleFileVisitor<Path> {
```
Comment (joanvr) [inline src/main/java/org/openrewrite/java/dependencies/github/ParseAdvisories.java:67]: why not make this a private static inner class? Also the naming is not very revealing... Maybe something related to Advisories?

**Review by joanvr — APPROVED** [review]

File: src/main/java/org/openrewrite/java/dependencies/github/ParseAdvisories.java (hunk 2 above)
Comment (timtebeek) [reply src/main/java/org/openrewrite/java/dependencies/github/ParseAdvisories.java]: Good suggestion, thanks. Adopted in https://github.com/openrewrite/rewrite-java-dependencies/pull/19/commits/a9850ed8bbaacdfc379e5e794a568d5ea36d375c

**Pull request merged**